In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv')

In [3]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [4]:
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

#### Qn 1 : What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [5]:
df.groupby('Item')['Y2014'].sum()

Item
Alcohol, Non-Food        2403.00
Alcoholic Beverages    102410.11
Animal Products         11935.65
Animal fats            209460.54
Apples and products      9499.23
                         ...    
Vegetables, Other      155038.96
Vegetal Products       107145.19
Wheat and products     232670.13
Wine                     4497.36
Yams                   200396.96
Name: Y2014, Length: 119, dtype: float64

In [6]:
df.groupby('Item')['Y2017'].sum()

Item
Alcohol, Non-Food        2348.00
Alcoholic Beverages     95581.06
Animal Products         11547.65
Animal fats            269617.53
Apples and products     10198.90
                         ...    
Vegetables, Other      157752.59
Vegetal Products       107655.20
Wheat and products     240047.62
Wine                     4178.02
Yams                   229174.59
Name: Y2017, Length: 119, dtype: float64

For Y2014 - 209460.54
For Y2017 - 269617.53

#### Qn 2. What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [7]:
#For standard deviation in year 2015 rounded off to 3 decimal places
round(df['Y2016'].std(),3)

1640.007

In [8]:
##For mean in year 2015 rounded off to 3 decimal places
round(df['Y2016'].mean(),3)

136.555

The closest matching option is 135.236 and 1603.404

#### Qn 3. What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [9]:
df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

Therefore there are 1535 missing data entries in the year 2016

In [10]:
df.isnull().sum() * 100 / len(df)

Area Code       0.000000
Area            0.000000
Item Code       0.000000
Item            0.000000
Element Code    0.000000
Element         0.000000
Unit            0.000000
Y2014           2.607354
Y2015           2.540078
Y2016           2.518747
Y2017           2.471162
Y2018           2.356300
dtype: float64

The percetage missing in year 2016 is 2.518 = 2.52

#### Qn 4. Which year had the highest correlation with ‘Element Code’? 

In [11]:
 df['Element Code']. corr(df['Y2014'])

0.024456819495222142

In [12]:
 df['Element Code']. corr(df['Y2015'])

0.02388919930164082

In [13]:
 df['Element Code']. corr(df['Y2016'])

0.023443997584961178

In [14]:
 df['Element Code']. corr(df['Y2017'])

0.024254007760045347

In [15]:
 df['Element Code']. corr(df['Y2018'])

0.02427857195351955

Highest correlation in the year 2014

In [16]:
df.corrwith(df['Element Code']).sort_values()

Item Code      -0.024683
Area Code      -0.000209
Y2016           0.023444
Y2015           0.023889
Y2017           0.024254
Y2018           0.024279
Y2014           0.024457
Element Code    1.000000
dtype: float64

#### Qn 5. What year has the highest sum of Import Quantity?

Hint- Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question

In [17]:
df['Y2014'].sum()

7965086.130000001

In [18]:
df['Y2015'].sum()

8032340.2299999995

In [19]:
df['Y2016'].sum()

8112472.649999999

In [20]:
df['Y2017'].sum()

8375729.189999999

In [21]:
df['Y2018'].sum()

8554629.96

#### Qn 6. What is the total number of the sum of Production in 2014?

Hint- Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question

In [22]:
df2 = (df.groupby('Element')['Y2014'].sum())

In [23]:
df2['Production']

1931287.7500000005

#### Qn 7. Which of these elements had the highest sum in 2018?

Hint-  Select columns ‘Y2018’ and ‘Element’, Perform a groupby operation on ‘Element’ on the selected dataframe and answer the  question.

In [24]:
(df.groupby('Element')['Y2018'].sum()).sort_values()

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

#### Qn 8. Which of these elements had the 3rd lowest sum in 2018?

Hint-  Select columns ‘Y2018’ and ‘Element’, Perform a groupby operation on ‘Element’ on the selected dataframe and answer the  question.  

In [25]:
(df.groupby('Element')['Y2018'].sum()).sort_values()

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

Protein supply quantity (g/capita/day) has third lowest sum

#### Qn 9. What is the total Import Quantity in Algeria in 2018?

In [26]:
df2 = df.groupby(['Area']).agg({'Y2014':'sum','Y2015':'sum','Y2016':'sum','Y2017':'sum','Y2018':'sum'})

In [27]:
df2.head()

Y2014      Y2015      Y2016      Y2017      Y2018
Area                                                               
Algeria       313375.33  324058.40  326007.41  325644.27  341729.98
Angola        208584.42  209565.67  214725.32  229159.57  235206.68
Benin         118050.79  108181.13  117963.59  124771.22  124960.70
Botswana       22339.92   23024.85   22970.65   22101.30   23129.08
Burkina Faso  105051.33  102701.22  107702.27  101855.07  112970.72

In [28]:
df2['Sum'] = df2['Y2014'] + df2['Y2015']+df2['Y2016']+df2['Y2017']+df2['Y2018']
df2

Y2014       Y2015       Y2016       Y2017  \
Area                                                                          
Algeria                       313375.33   324058.40   326007.41   325644.27   
Angola                        208584.42   209565.67   214725.32   229159.57   
Benin                         118050.79   108181.13   117963.59   124771.22   
Botswana                       22339.92    23024.85    22970.65    22101.30   
Burkina Faso                  105051.33   102701.22   107702.27   101855.07   
Cabo Verde                     14576.94    14538.93    14569.40    14650.74   
Cameroon                      217630.28   225220.72   231714.36   232030.43   
Central African Republic       28739.02    28885.34    29454.54    29937.00   
Chad                           65940.17    65835.88    72673.76    71594.68   
Comoros                           32.00       41.34       54.91       59.84   
Congo                          38699.26    40255.51    40592.66    41181.68   
C�te d'Ivoire                 195982.84   211591.29   211439.20   224599.01   
Djibouti                       21332.28    23690.16    21783.91    22729.91   
Egypt                         849192.30   868218.73   858568.63   866379.92   
Eswatini                       53285.75    54072.07    54025.21    54343.33   
Ethiopia                      415935.50   429139.33   444567.57   448683.76   
Ethiopia PDR                       0.00        0.00        0.00        0.00   
Gabon                          28126.31    28367.30    28129.95    27979.64   
Gambia                         19607.87    19665.39    21758.15    23154.18   
Ghana                         308328.59   311092.13   314320.45   337599.06   
Guinea                         91213.40    92803.39    95450.24    98138.87   
Guinea-Bissau                  17708.67    17990.06    18524.83    19102.77   
Kenya                         255211.01   265506.19   260257.56   264660.66   
Lesotho                        19125.89    17965.84    17258.96    21267.96   
Liberia                        27898.35    28698.10    29746.68    29342.20   
Madagascar                    128985.32   126674.90   129010.25   131197.73   
Malawi                        173454.86   167989.09   164970.42   181098.71   
Mali                          123038.75   132456.58   147324.78   149928.33   
Mauritania                    109557.63    61909.99    79872.42   156665.46   
Mauritius                      50898.12    53372.42    49272.41    51114.83   
Morocco                       375977.76   412473.80   367884.18   388495.36   
Mozambique                    160757.87   158231.17   159924.41   161407.98   
Namibia                        32026.42    30687.90    30389.07    29874.89   
Niger                         108236.65   117834.65   123601.88   126707.58   
Nigeria                      1399542.19  1414362.83  1482945.71  1483268.23   
Rwanda                         70001.38    70138.83    70921.76    73663.69   
Sao Tome and Principe          12458.64    12386.93    12647.56    12662.63   
Senegal                        78533.93    91939.14    89017.72    95681.15   
Seychelles                       305.20      358.20      416.32      442.34   
Sierra Leone                   62570.89    62523.14    52767.72    55311.33   
South Africa                  497289.52   468352.18   462224.16   517590.54   
Sudan                         247577.56   230652.54   253197.29   239931.92   
Sudan (former)                     0.00        0.00        0.00        0.00   
Togo                           47618.05    47834.59    47930.75    49841.88   
Tunisia                       117029.33   128211.41   123778.86   124167.20   
Uganda                        218142.33   220615.72   202981.34   213950.38   
United Republic of Tanzania   337430.64   341969.46   333765.86   322616.85   
Zambia                         99318.53    96214.00    97728.69   103223.77   
Zimbabwe                       78366.34    74041.79    75638.88    75919.34   

    

#### Qn 10. What is the total number of unique countries in the dataset?

In [29]:
df['Area'].nunique()

49